build.gradle

```diff
@@ -63,6 +63,25 @@ test {
 
 gradlePlugin {
     testSourceSets sourceSets.functionalTest, sourceSets.samples
+
+    plugins {
+        teamcityServerPlugin {
+            id = 'com.github.rodm.teamcity-server'
+            implementationClass = 'com.github.rodm.teamcity.TeamCityServerPlugin'
+        }
+        teamcityAgentPlugin {
+            id = 'com.github.rodm.teamcity-agent'
+            implementationClass = 'com.github.rodm.teamcity.TeamCityAgentPlugin'
+        }
+        teamcityCommonPlugin {
+            id = 'com.github.rodm.teamcity-common'
+            implementationClass = 'com.github.rodm.teamcity.TeamCityCommonPlugin'
+        }
+        teamcityEnvironmentsPlugin {
+            id = 'com.github.rodm.teamcity-environments'
+            implementationClass = 'com.github.rodm.teamcity.TeamCityEnvironmentsPlugin'
+        }
+    }
```

Comment:
@satamas I don't understand why this change is needed based on the comment can you explain the problem it fixes?

Reply:
When I was trying to test my plugin locally I faced a problem that it's impossible to load the plugin easily from the maven local. It was possible only if the plugin artifact was added to the buildScript as a dependency using full maven coordinates. With these changes, Gradle adds all required files, and a plugin from the local repo can be loaded by id in the same way as a plugin from the Gradle plugin portal.

Reply:
Ok I understand, the way I test the plugin is by adding the gradle plugin project using the 'includeBuild' function in a teamcity plugin project, I'll add a section on testing to the readme.

Reply:
I've found the block I'm talking about: https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers. Without my changes, there was no plugin markers in the local repo as far as I understand